### PR TITLE
Add initial support for finalizeImage

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Add option to provide a final script that modifies the chroot before creating an image

###### Change Log  
- Add option to provide a final script that modifies the chroot before creating an image

###### Does this affect the toolchain?  
**NO**

###### Associated issues  

- #[41439909](https://microsoft.visualstudio.com/OS/_workitems/edit/41439909)

###### Test Methodology

- Functional testing done: tested path with different number of scripts (no script, 1 script, multiple script paths), with and without args

### CODE REVIEW

**Comment by dmcilvaney** [comment]
We should have a section in the docs in `formats/imageconfig.md` for this.

File: toolkit/tools/imager/imager.go [hunk 1]
```diff
@@ -577,6 +593,13 @@ func buildImage(mountPointMap, mountPointToFsTypeMap, mountPointToMountArgsMap,
 		}
 	}
 
+	// Run finalize image scripts from within the installroot chroot
```
Comment (dmcilvaney) [inline toolkit/tools/imager/imager.go:596]: Curious if we should limit this to non-rootfs type images... I vaguely recall having misgivings about allowing this for non-image/disk based install scenarios but I don't recall what they were.

Any thoughts?

Comment (christopherco) [reply toolkit/tools/imager/imager.go:596]: Only issue I could think of is possible confusion with the post-install scripts field for rootfs only.

Maybe something to do with the bespoke delta update path, which would fall under the category of non-image-based install scenarios?

**Review by christopherco — APPROVED** [review]

File: toolkit/tools/imager/imager.go (hunk 1 above)
Comment (christopherco) [reply toolkit/tools/imager/imager.go:596]: Spoke with @dmcilvaney and we both can't see any issues with the non-image path

**Pull request merged**